### PR TITLE
Add dynamic Management Pane UI

### PR DESCRIPTION
**Pull request by tangledbytes**
**Description**

This PR adds the following:
1. Dynamic Management Page UI - `/manamgement/:mesh-name` will render a dynamic form for the `:mesh-name` in the service mesh
2. Support for Dynamic Routes on the golang server - Golang server now supports generic Next.js dynamic paths
3. Support for pattern files as JSON - `POST` and `DELETE` requests on `/api/experimental/pattern/deploy` now accept header `ContentType: application/json` which will accept patterns as JSON.

**Relevant Screenshot**
Screenshot_20210614_205428

**Notes for Reviewers**
The new UI is not exposed on the navigator and needs to be accessed manually.

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Ayush-Khandelwal-007** [comment]
I believe all logical job is done very well by @utkarsh-pro now a bit of styling will be required.Right?

**Comment by Ayush-Khandelwal-007** [comment]
Hey @utkarsh-pro does the fields like Istio version , and all gives the list of all the available options or it is still requires to be typed in?

**Comment by utkarsh-pro** [comment]
> Hey @utkarsh-pro does the fields like Istio version , and all gives the list of all the available options or it is still requires to be typed in?

@Ayush-Khandelwal-007 actually that will be handled on the adapter side and not in the UI. The UI will automatically change as soon as we make changes in the adapter.

**Comment by Ayush-Khandelwal-007** [comment]
> > Hey @utkarsh-pro does the fields like Istio version , and all gives the list of all the available options or it is still requires to be typed in?
> 
> @Ayush-Khandelwal-007 actually that will be handled on the adapter side and not in the UI. The UI will automatically change as soon as we make changes in the adapter.

😅Who will be the most appropriate one to do that task? @leecalcote @utkarsh-pro

**Comment by leecalcote** [comment]
@Ayush-Khandelwal-007 great question. This falls inline with our prior discussions on revamping Meshery's UI, starting with the management pages. @Ayush-Khandelwal-007 this might be you. 🙂

File: handlers/oam_handler.go [hunk 1]
```diff
@@ -57,6 +58,15 @@ func (h *Handler) PatternFileHandler(
 		return
 	}
 
+	if r.Header.Get("Content-Type") == "application/json" {
+		body, err = yaml.JSONToYAML(body)
+		if err != nil {
+			rw.WriteHeader(http.StatusBadRequest)
+			fmt.Fprintf(rw, "failed to parse to PatternFile: %s", err)
```
Comment (leecalcote) [inline handlers/oam_handler.go:65]: @utkarsh-pro are we ready for the new error code system?

(not correct, but an example of sorts)
```return errors.New(ErrHTTPStatusBadRequest, errors.Alert, []string{"failed to parse to PatternFile: %s", err.Error()}, []string{err.Error()}, []string{}, []string{})```

**Review by leecalcote — APPROVED** [review]

**Comment by adithyaakrishna** [comment]
Hey @utkarsh-pro 
This PR adds the New management Thing right? For which I talked about improving the UI thing & adding a modal based one instead of dropdown thing

**Comment by leecalcote** [comment]
@adithyaakrishna yes, indeed. Much focus and help needed here.

**Pull request merged**